sgkit/display.py

```diff
@@ -0,0 +1,192 @@
+from typing import Any, Tuple
+
+import pandas as pd
+import xarray as xr
+
+from .typing import ArrayLike
+
+
+class GenotypeDisplay:
+    """
+    A printable object to display genotype information.
+    """
+
+    def __init__(
+        self,
+        df: pd.DataFrame,
+        shape: Tuple[int, int],
+        max_variants: int,
+        max_samples: int,
+    ):
+        self.df = df
+        self.shape = shape
+        self.max_variants = max_variants
+        self.max_samples = max_samples
+        self.pd_options = [
+            "display.max_rows",
+            self.max_variants,
+            "display.min_rows",
+            self.max_variants,
+            "display.max_columns",
+            self.max_samples,
+            "display.show_dimensions",
+            False,
+        ]
+
+    def __repr__(self) -> Any:
+        with pd.option_context(*self.pd_options):
+            if (
+                len(self.df) > self.max_variants
+                or len(self.df.columns) > self.max_samples
+            ):
+                return (
+                    self.df.__repr__()
+                    + f"\n\n[{self.shape[0]} rows x {self.shape[1]} columns]"
+                )
+            return self.df.__repr__()
+
+    def _repr_html_(self) -> Any:
+        with pd.option_context(*self.pd_options):
+            if (
+                len(self.df) > self.max_variants
+                or len(self.df.columns) > self.max_samples
+            ):
+                return self.df._repr_html_().replace(
+                    "</div>",
+                    f"<p>{self.shape[0]} rows x {self.shape[1]} columns</p></div>",
+                )
+            return self.df._repr_html_()
+
+
+def _truncate(
+    arr: xr.DataArray,
```

Comment:
Should we have a preference for qualified type annotations like this vs imports of the type at the top level?  I'm not aware of any meaningful difference, but I've been doing the latter everywhere.  I like the qualified version to avoid DataFrame conflicts so perhaps we should make that a standard.

Reply:
I don't have a strong view on this.